data-report-commons/src/main/java/commons/ViewCompiler.java

```diff
@@ -21,17 +23,30 @@ public ViewCompiler(InputStream inputStream) {
         RDFDataMgr.read(model, inputStream, Lang.JSONLD);
     }
 
-    public Model extractPublicationView() {
-        var query = IoUtils.stringFromResources(NVA_QUERY);
+    public Model extractView() {
+        if (isPublication(model)) {
+            return extractPublicationView();
+        } else {
+            return extractNviCandidateView();
+        }
+    }
+
+    private Model extractNviCandidateView() {
+        var query = IoUtils.stringFromResources(NVI_QUERY);
         try (var queryExecution = QueryExecutionFactory.create(query, model)) {
             return queryExecution.execConstruct();
         }
     }
 
-    public Model extractNviCandidateView() {
-        var query = IoUtils.stringFromResources(NVI_QUERY);
+    private Model extractPublicationView() {
+        var query = IoUtils.stringFromResources(NVA_QUERY);
         try (var queryExecution = QueryExecutionFactory.create(query, model)) {
             return queryExecution.execConstruct();
         }
     }
+
+    private boolean isPublication(Model model) {
+        var publicationType = model.createResource(PUBLICATION);
+        return model.listResourcesWithProperty(RDF.type, publicationType).hasNext();
```

Comment:
This method is a bit sketchy, better idea @brinxmat?

Reply:
See latest commit: 
[Use id to verify resource type](https://github.com/BIBSYSDEV/nva-data-report-api/pull/146/commits/9514c156c2306b56183af650e0d458f75f5a626e)
I think this is much better